compiler/lookups.nim

```diff
@@ -295,7 +295,8 @@ proc ensureNoMissingOrUnusedSymbols(c: PContext; scope: PScope) =
         # maybe they can be made skGenericParam as well.
         if s.typ != nil and tfImplicitTypeParam notin s.typ.flags and
            s.typ.kind != tyGenericParam:
-          unusedSyms.add (s, toFileLineCol(c.config, s.info))
+          if s.name.s.startsWith("_`gensym") == false:
```

Comment:
That's a hack. Instead you should add `sfUsed` to the gensym'ed symbol or something like that.

Reply:
okay, I'll dig deeper since no useable flags here.

Reply:
I dont track further enough "where the symbol lose `sfGenSym` flag", instead I fix `isDiscardUnderscore`, it even dont add to declaration now.